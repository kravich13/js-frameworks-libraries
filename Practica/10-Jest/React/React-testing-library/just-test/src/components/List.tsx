import React, { useState } from 'react'

export const List: React.FC = () => {
  const [news, setNews] = useState<any[]>([])
  const [error, setError] = useState<any | null>(null)

  const handleFetch = async () => {
    try {
      const result: Response = await fetch(
        'https://jsonplaceholder.typicode.com/posts?post=5'
      )
      const jsonRes: any[] = await result.json()
      setNews(jsonRes)
    } catch (err) {
      setError(err)
    }
  }
  return (
    <div>
      <button data-testid="posts-click" onClick={handleFetch}>
        Новости
      </button>

      {error && <span>Что-то пошло не так...</span>}

      {news.map(({ title, id }) => {
        return <li key={id}>{title}</li>
      })}
    </div>
  )
}
