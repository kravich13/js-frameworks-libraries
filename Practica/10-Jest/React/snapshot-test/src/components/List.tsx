import React from 'react'

export const List: React.FC<any> = ({ list }) => {
  if (list.length < 0) return <div>No items in list</div>
  return (
    <ol>
      {list.map((title: string, index: number) => {
        return <li key={index}>{title}</li>
      })}
    </ol>
  )
}
