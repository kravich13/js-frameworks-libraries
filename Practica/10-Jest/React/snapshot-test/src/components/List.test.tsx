import renderer from 'react-test-renderer'
import { List } from './List'

it('renders when there are not items', () => {
  const tree = renderer.create(<List list={[]} />).toJSON()

  expect(tree).toMatchSnapshot()
})

it('renders when there are three items', () => {
  const tree = renderer
    .create(<List list={['Vlad', 'Max', 'Kravich']} />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})
