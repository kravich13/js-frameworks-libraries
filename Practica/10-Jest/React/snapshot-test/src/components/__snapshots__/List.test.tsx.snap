// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when there are not items 1`] = `<ol />`;

exports[`renders when there are three items 1`] = `
<ol>
  <li>
    Vlad
  </li>
  <li>
    Max
  </li>
  <li>
    Kravich
  </li>
</ol>
`;
